src/error.rs

```diff
@@ -1,6 +1,7 @@
 use failure::{self, Context, Fail};

```

Comment:
I'm not sure how involved that change would be, since there's a lot of error handling code already. Once we have functional multi-server code, I'll see if we can make the switch.